packages/jsii/lib/assembler.ts

```diff
@@ -148,11 +146,38 @@ export class Assembler implements Emitter {
      * Defer checks for after the program has been entirely processed; useful for verifying type references that may not
      * have been discovered yet, and verifying properties about them.
      *
+     * @param fqn FQN of the current type.
+     * @param deps List of FQNs of types this callback depends on. All deferreds for all
      * @param cb the function to be called in a deferred way. It will be bound with ``this``, so it can depend on using
      *           ``this``.
      */
-    private _defer(cb: () => void) {
-        this._deferred.push(cb.bind(this));
+    private _defer(fqn: string, depFqns: string[], cb: () => void) {
+        this._deferred.push({ fqn, depFqns, cb: cb.bind(this) });
+    }
+
+    /**
+     * Defer a callback until a (set of) types are available
+     *
+     * This is a helper function around _defer() which encapsulates the _dereference
+     * action (which is basically the majority use case for _defer anyway).
+     *
+     * Will not invoke the function with any 'undefined's; an error will already have been emitted in
+     * that case anyway.
+     */
+    // tslint:disable-next-line:max-line-length
+    private _deferUntilTypeAvailable(fqn: string, baseTypes: NamedTypeReference[], referencingNode: ts.Node, cb: (...xs: spec.Type[]) => void) {
```

Comment:
Should it not be `_deferUntilType+s+Available`?
I would also like it if the doc had `@param`s.